RoaringBitmap/src/main/java/org/roaringbitmap/BitSetUtil.java

```diff
@@ -14,14 +14,75 @@
  *
  */
 public class BitSetUtil {
-  // todo: add a method to convert a RoaringBitmap to a BitSet using BitSet.valueOf
 
   // a block consists has a maximum of 1024 words, each representing 64 bits,
   // thus representing at maximum 65536 bits
   public static final int BLOCK_LENGTH = BitmapContainer.MAX_CAPACITY / Long.SIZE; //
   // 64-bit
   // word
 
+  /**
+   * Convert a {@link RoaringBitmap} to a {@link BitSet}.
+   * <p>
+   * Equivalent to calling {@code BitSet.valueOf(BitSetUtil.toLongArray(bitmap))}.
+   */
+  public static BitSet bitsetOf(RoaringBitmap bitmap) {
+    return BitSet.valueOf(toLongArray(bitmap));
+  }
+
+  /**
+   * Returns an array of long, given a {@link RoaringBitmap}.
+   * <p>
+   * See {@link BitSet#toByteArray()}.
+   */
+  public static byte[] toByteArray(RoaringBitmap bitmap) {
+    long[] words = toLongArray(bitmap);
+    ByteBuffer buffer = ByteBuffer.allocate(words.length * Long.SIZE)
+            .order(ByteOrder.LITTLE_ENDIAN);
+    buffer.asLongBuffer().put(words);
+    return buffer.array();
+  }
+
+  /**
+   * Returns an array of little-endian ordered bytes, given a {@link RoaringBitmap}.
+   * <p>
+   * See {@link BitSet#toByteArray()}.
+   */
+  public static long[] toLongArray(RoaringBitmap bitmap) {
+    if (bitmap.isEmpty()) {
+      return new long[0];
+    }
+
+    int lastBit = Math.max(bitmap.last(), Long.SIZE);
```

Comment:
`bitmap.last()` can produce a negative `int` for bitmaps with values >= 2^31, so it should be masked with `0xFFFF_FFFFL` and the result should be a `long`, with minor consequences for the arithmetic below.

Reply:
Sound like I can address both comments by just bailing if `bitmap.last() < 0`?

Reply:
It depends. A long[] has enough bits to store any RoaringBitmap, so it could produce a result here, in which case your arithmetic needs to handle values in the [2^31, 2^32) range. Or you could just check and bail and someone else can add support for larger bitmaps if they really want it down the line.